src/geode/geometry/basic_objects/tetrahedron.cpp

```diff
@@ -79,6 +82,35 @@ namespace geode
         return bbox;
     }
 
+    template < typename PointType >
+    bool GenericTetrahedron< PointType >::is_degenerated() const
+    {
+        for( const auto [vertex0, vertex1, vertex2] : tetrahedron_facet_vertex )
+        {
+            const Point3D& point0 = vertices_.at( vertex0 );
+            const Point3D& point1 = vertices_.at( vertex1 );
+            const Point3D& point2 = vertices_.at( vertex2 );
+            const Triangle3D triangle{ point0, point1, point2 };
+            if( triangle.is_degenerated() )
+            {
+                return true;
+            }
+            for( const auto vertex3 : LRange{ 4 } )
+            {
+                if( vertex3 == vertex0 || vertex3 == vertex1
+                    || vertex3 == vertex2 )
+                {
+                    continue;
+                }
+                const Point3D& point3 = vertices_.at( vertex3 );
+                return std::get< 0 >(
+                           point_plane_distance( point3, Plane{ triangle } ) )
+                       <= GLOBAL_EPSILON;
```

Comment:
not a return here. but if <= GLOBAL_EPSILON then return true, else go for the next tetrahedron_facet

Reply:
Or you do not have to iterates in the 4 facets, since you will directly have an answer on the first tested facet

Reply:
Might be enough... If it is degen then the simplex is degen, is not the distance once should be enough to determine since we are in a simplex. What do you think?

Reply:
So remove the main for loop?